cylc/flow/cycling/iso8601.py

```diff
@@ -51,14 +57,16 @@
 class SuiteSpecifics:
 
     """Store suite-setup-specific constants and utilities here."""
-    ASSUMED_TIME_ZONE = None
-    DUMP_FORMAT = None
-    NUM_EXPANDED_YEAR_DIGITS = None
-    abbrev_util = None
-    interval_parser = None
-    point_parser = None
-    recurrence_parser = None
-    iso8601_parsers = None
+    ASSUMED_TIME_ZONE: Optional[Tuple[int, int]] = None
+    DUMP_FORMAT: Optional[str] = None
+    NUM_EXPANDED_YEAR_DIGITS: Optional[int] = None
+    abbrev_util: Optional[CylcTimeParser] = None
+    interval_parser: Optional['DurationParser'] = None
+    point_parser: Optional['TimePointParser'] = None
+    recurrence_parser: Optional['TimeRecurrenceParser'] = None
+    iso8601_parsers: Optional[
+        Tuple['DurationParser', 'TimePointParser', 'TimeRecurrenceParser']
+    ] = None
```

Comment:
These lines are causing mypy errors for me:

```
cylc/flow/cycling/iso8601.py:683: error: Item "None" of "Optional[Any]" has no attribute "parse"
cylc/flow/cycling/iso8601.py:691: error: Item "None" of "Optional[Any]" has no attribute "parse"
cylc/flow/cycling/iso8601.py:693: error: Item "None" of "Optional[Any]" has no attribute "parse"
cylc/flow/cycling/iso8601.py:720: error: Item "None" of "Optional[Any]" has no attribute "parse"
cylc/flow/cycling/iso8601.py:751: error: Incompatible types in assignment (expression has type "None", variable has type "str")
cylc/flow/cycling/iso8601.py:784: error: Item "None" of "Optional[Any]" has no attribute "parse"
cylc/flow/cycling/iso8601.py:793: error: "str" has no attribute "hour_of_day"
cylc/flow/cycling/iso8601.py:794: error: "str" has no attribute "minute_of_hour"
cylc/flow/cycling/iso8601.py:795: error: "str" has no attribute "second_of_minute"
cylc/flow/cycling/iso8601.py:803: error: "str" has no attribute "month_of_year"
cylc/flow/cycling/iso8601.py:804: error: "str" has no attribute "day_of_month"
cylc/flow/cycling/iso8601.py:807: error: "str" has no attribute "day_of_month"
```

Reply:
Hence the
> By the way, it looks like mypy isn't going to be happy about the way the cycle point stuff is done...

I'll see what I can do, may be worth cutting this commit from the PR and tackling later

Reply:
For some reason, `point_parser: 'TimePointParser' = None` works. Latest force push should have got rid of the errors

Reply:
A little strange, I guess this is because isodatetime doesn't have type attrs or a mypy "stub" so it doesn't know what to do about this external import? Dunno.

Rebased onto master and ran mypy, no errors, thanks.